### PR TITLE
#598 PaymentMethods API and StoredPaymentMethod.

### PR DESCRIPTION
**Pull request by criske**
PR for #598
- left puzzle for PaymentMethods#ofWallet

### CODE REVIEW

File: self-core-impl/src/main/java/com/selfxdsd/core/projects/StoredPaymentMethod.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Override
+    public Wallet wallet() {
+        return this.wallet;
+    }
+
+    @Override
+    public boolean active() {
+        return this.equals(this.storage.paymentMethods()
```
Comment (criske) [inline self-core-impl/src/main/java/com/selfxdsd/core/projects/StoredPaymentMethod.java:80]: @amihaiemil Should we provide this boolean in constructor instead, might be more efficient.

Comment (amihaiemil) [reply self-core-impl/src/main/java/com/selfxdsd/core/projects/StoredPaymentMethod.java:80]: @criske yes, definetely :D we do the same for wallets and PayoutMethods

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@criske looks nice. Just one comment, please provide the ``active`` flag in the constructor.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Pull request merged**